File: Бойко Дмитро/Лабораторна робота №1/Bill.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab_1
{
    class Bill
    {
        public double LimitingAmount { get; set; }
        public double CurrentDebt { get; private set; }

        public Bill(double limitingAmount)
        {
            LimitingAmount = limitingAmount;
            CurrentDebt = 0.0;
        }
        public bool Check(double amount)
        {
            return CurrentDebt + amount <= LimitingAmount;
        }

        public void Add(double amount)
        {
            if (Check(amount))
            {
                CurrentDebt += amount;
            }
            else
            {
                Console.WriteLine("Ліміт перевищено. Неможливо додати більше до рахунку.");
            }
        }

        public void Pay(double amount)
        {
            if (amount <= CurrentDebt)
            {
                CurrentDebt -= amount;
            }
            else
            {
                Console.WriteLine("Сума платежу перевищує поточну заборгованість.");
            }
        }

        public void ChangeTheLimit(double amount)
        {
            LimitingAmount = amount;
        }
    }
}

File: Пащук Нікіта/Lab1/Lab_1/Bill.py
class Bill:
    def __init__(self, limiting_amount: float, current_debt=0):
        self.limitingAmount = limiting_amount
        self.current_debt = current_debt

    def check(self, current_debt: float):
        if self.current_debt > self.limitingAmount:
            return False
        else:
            return True

    def add(self, amount: float):
        if self.current_debt + amount >= self.limitingAmount:
            return "Cannot add: limit exceeded"
        else:
            self.current_debt += amount
        return f"Customer has added {amount} to his debt, current debt is {self.current_debt}"

    def pay(self, amount: float):
        if self.current_debt - amount <= 0:
            return "Cannot pay: negative value"
        else:
            self.current_debt -= amount
        return f'Customer has payed {amount} for his debt, current debt is {self.current_debt}'

    def check_limit(self):
        return f"Your current limit is {self.limitingAmount}"

    def change_the_limit(self, amount: float):
        self.limitingAmount = amount
        return f'Your new limit is {self.limitingAmount}'
